packages/engine/src/datastore/table/pool/message.rs

```diff
@@ -51,6 +50,51 @@ impl MessagePool {
     pub fn push(&mut self, batch: MessageBatch) {
         self.batches.push(Arc::new(RwLock::new(batch)))
     }
+
+    /// Clears all message columns in the pool and resizes them as necessary to accommodate new
+    /// messages according to the provided `agent_proxies`.
+    ///
+    /// This can result in the number of batches being changed if more/less are now needed.
+    ///
+    /// # Panics
+    ///
+    /// If the message pool contains batches, which are currently borrowed.
+    pub fn reset(
+        &mut self,
+        agent_proxies: &PoolReadProxy<AgentBatch>,
+        sim_config: &SimRunConfig,
+    ) -> Result<()> {
+        // Reversing sequence to remove from the back if there are fewer agent batches than message
+        // batches
+        for batch_index in (0..self.len()).rev() {
+            if let Some(dynamic_batch) = agent_proxies.batch(batch_index) {
+                let mut write_proxy = BatchWriteProxy::new(&self.batches[batch_index]).expect(
+                    "Failed to remove Batch at index {index}, other Arcs to the Batch existed",
+                );
```

Comment:
I'm not sure we should be overriding the error message already there. I appreciate this will be better solved when we use the new error library however you're adding context here that's buried beneath an abstraction (as in about the Arc and such), if we change the Proxy implementation the error message is wrong.

Maybe just do a `map_err` with a format string to add the index?
```suggestion
                let mut write_proxy = BatchWriteProxy::new(&self.batches[batch_index]).wrap_err(|err|
                    format!("Failed to remove Batch at index {index}: {err}"),
                );
```

Pretty sure that doesn't satisfy rustfmt

Reply:
The reason why I changed it to panic is, that `self.remove()` will panic with the same parameters. It feels weird, one arm is panicking and one is erroring. The if-clause is pretty unrelated to the actual error, so both arms should behave the same.
I agree, however, that the error message can be expanded.

Reply:
Gotcha, apologies, I forgot expect does actually wrap the underlying error rather than override it. 

Lets perhaps remove the specific bit about Arcs to the batch then and just say `Failed to remove Batch at index {index}`

Reply:
```suggestion
                let mut write_proxy = BatchWriteProxy::new(&self.batches[batch_index]).expect(
                    "Failed to remove Batch at index {index}",
                );
```

Reply:
Actually, `expect` can't take `{index}` as it's not a macro. Clippy will suggest to `unwrap_or_else` and `panic`. That's probably the best way to handle this.